File: seek_hit/Zadatak 3/task_3.py
import random

#Klasa avion
class Airplane:
    def __init__(self, id):
        self.id = id
        self.park_id = 0

    def print_plane(self):
        print ("Plane "+ str(self.id) + " has parking slot " + str(self.park_id))

#Klasa parking
class Parking:
    def __init__(self, id):
        self.id = id
        self.status = "empty"

    def print_parking(self):
        print ("Parking spot " + str(self.id) + " is " + self.status)

#Parkiranje aviona
def park_plane(airplane_list, parking_list, houndred_list):
    for plane in airplane_list:
        if plane.park_id == 0:
            assign_parking(plane, parking_list, houndred_list)
            plane.print_plane()
        else:
            print ("Airlplane " + str(plane.id) + " already has parking slot " + str(plane.park_id))

#Dodjeljivanje random broja parkinga avionu
def assign_parking (plane, parking_list, houndred_list):
    temp =  random.choice(houndred_list)
    index = houndred_list.index(temp)
    houndred_list.pop(index)

    #Provjera da li su parkirna mista zauzeta
    if (parking_list[temp].status == "taken"):
        assign_parking(plane, parking_list, houndred_list)
        return
    plane.park_id = parking_list[temp].id
    parking_list[temp].status = "taken"

#main
if __name__ == "__main__":
    airplane_list = []
    parking_list = []
    hundred_list = []

    #Dodavanje liste objekata
    for i in range (0,80):
        plane = Airplane(i+1)
        airplane_list.append(plane)

    #Dodavanje liste objekata i liste intigera
    for i in range (0,100):
        park = Parking(i+1)
        parking_list.append(park)
        hundred_list.append(i)

    #Dodjeljivanje random broja iz liste houndred avionu
    park_plane(airplane_list, parking_list, hundred_list)

    #Druga provjera da li je parking zauzet
    park_plane(airplane_list, parking_list, hundred_list)

    for park in parking_list:
        park.print_parking()

    #Ispis praznih parkinga
    print("Empty parking slots are: " + str(hundred_list))

File: seek_hit/Zadatak 3/task_3.lua
--Funkcija za inicijaliziranje objekta Airplane + ispis
function airplane_init(id)
  local Airplane = {}
    Airplane.id = id
    Airplane.park_id = 0
  function Airplane:report()
    print("Plane " .. Airplane.id .. " has parking slot " .. Airplane.park_id)
  end
    
  return Airplane
end
  
  
 --Funkcija za inicijaliziranje objekta Parking + ispis 
function parking_init(id)  
  local Parking = {}
    Parking.id = id
    Parking.status = "epty"
    
  function Parking:report()
    print("Parking spot " .. Parking.id .. " is " .. Parking.status)
  end

  return Parking
end


--Funkcija za parkiranje aviona
--U .py fileu sam ovde ima provjeru, a kako se u funkciji assign_parking vec vrsi provjera ova je bila visak
function park_plane(airplane_list, parking_list)
  for i, plane in ipairs(airplane_list) do
     assign_parking(plane, parking_list)
      print(plane:report())
  end
end  


--Funkcija za dodjeljivanje random mjesta avionu
--U .py file-u nije bilo moguce generirati random broj koji je vec bio definiran
--U ovoj funkciji je to moguce i ako je izgeneriran vec pridodati broj ispisat ce se poruka
function assign_parking(plane, parking_list)
  temp = math.random(1,100)
  parking = parking_list[temp]
    --Provjera da li su parkirna mista zauzeta
    if (parking.status == "taken") then
        print ("Parking " .. temp .. " has already been assigned")
        assign_parking(plane, parking_list)

  else
      plane.park_id = parking_list[temp].id
      parking.status = "taken"
    end
end


--Main
airplane_list = {}
parking_list = {}

for i = 0, 80 do
  airplane_list[i] = airplane_init(i)  
end

for i = 0, 100 do
  parking_list[i] = parking_init(i)  
end

park_plane(airplane_list, parking_list)

for i, park in ipairs(parking_list) do
   print (park:report())
end










